src/lib/middleware/audit-middleware.ts

```diff
@@ -0,0 +1,21 @@
+import type { IUnleashConfig } from '../types';
+import type { IApiRequest, IAuthRequest } from '../routes/unleash-types';
+import { extractAuditInfo } from '../util';
+
+export const auditAccessMiddleware = ({
+    getLogger,
+}: Pick<IUnleashConfig, 'getLogger'>): any => {
+    const logger = getLogger('/middleware/audit-middleware.ts');
+    return (req: IAuthRequest | IApiRequest, _res, next) => {
+        if (!req.user) {
+            logger.info('Could not find user');
+        } else {
```

Comment:
Do we want to log everytime we don't find a user?
```suggestion
        if (req.user) {
```

Reply:
Since this middleware is only mounted on `/api/admin` not finding a user might actually be an error.